datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -115,6 +106,14 @@ struct GroupedHashAggregateStreamInner {
     indices: [Vec<Range<usize>>; 2],
 }
 
+#[derive(Debug)]
+/// tracks what phase the aggregation is in
+enum ExecutionState {
```

Comment:
This used to be tracked using several multi-level `match` statement and a `fused` inner stream. Now it is represented explicitly in this stream